Tests/ApolloTests/RequestCreatorTests.swift

```diff
@@ -283,4 +284,135 @@ Bravo file content.
       self.checkString(stringToCompare, includes: endString)
     }
   }
+
+  func testRequestBodyWithApolloRequestCreator() {
+    let query = HeroNameQuery()
+    let req = apolloRequestCreator.requestBody(for: query, sendOperationIdentifiers: false)
+
+    XCTAssertEqual(query.queryDocument, req["query"] as? String)
+  }
+
+  // MARK: - Custom request creator tests
+
+  func testSingleFileWithCustomRequestCreator() throws {
+    let alphaFileUrl = self.fileURLForFile(named: "a", extension: "txt")
+
+    let alphaFile = GraphQLFile(fieldName: "upload",
+                                originalName: "a.txt",
+                                mimeType: "text/plain",
+                                fileURL: alphaFileUrl)
+
+    let data = try customRequestCreator.requestMultipartFormData(
+      for: HeroNameQuery(),
+      files: [alphaFile!],
+      sendOperationIdentifiers: false,
+      serializationFormat: JSONSerializationFormat.self,
+      manualBoundary: "TEST.BOUNDARY"
+    )
+
+    let stringToCompare = try self.string(from: data)
+
+    print(stringToCompare)
+
+    if #available(iOS 11, macOS 13, tvOS 11, watchOS 4, *) {
+      let expectedString = """
+--TEST.BOUNDARY
+Content-Disposition: form-data; name="test_variables"
+
+{"episode":null}
+--TEST.BOUNDARY
+Content-Disposition: form-data; name="test_query"
+
+query HeroName($episode: Episode) { hero(episode: $episode) { __typename name } }
+--TEST.BOUNDARY
+Content-Disposition: form-data; name="test_operationName"
+
+HeroName
+--TEST.BOUNDARY
+Content-Disposition: form-data; name="upload"; filename="a.txt"
+Content-Type: text/plain
+
+Alpha file content.
+
+--TEST.BOUNDARY--
+"""
+      XCTAssertEqual(stringToCompare, expectedString)
+    } else {
+      // Operation parameters may be in weird order, so let's at least check that the files and single parameter got encoded properly.
+      let expectedEndString = """
+--TEST.BOUNDARY
+Content-Disposition: form-data; name="map"
+
+{"0":["variables.upload"]}
+--TEST.BOUNDARY
+Content-Disposition: form-data; name="0"; filename="a.txt"
+Content-Type: text/plain
+
+Alpha file content.
+
+--TEST.BOUNDARY--
+"""
+      self.checkString(stringToCompare, includes: expectedEndString)
+    }
+  }
+
+  func testRequestBodyWithCustomRequestCreator() {
+    let query = HeroNameQuery()
+    let req = customRequestCreator.requestBody(for: query, sendOperationIdentifiers: false)
+
+    XCTAssertEqual(query.queryDocument, req["test_query"] as? String)
+  }
+}
+
+fileprivate struct TestRequestCreator: RequestCreator {
```

Comment:
Hm...I'm torn on whether to suggest pulling this out into a separate file - though if you do I'd probably call it something more like `TestCustomRequestCreator` to clarify that it's not the request creator that should be used for all tests.

Reply:
Will pull out. I was a little should I do or not. 
Some people want to keep in file other want it out 🤷‍♂️

I like the second one.